### PR TITLE
[AIRFLOW-4763] Allow list in DockerOperator.command

### PR DESCRIPTION
**Pull request by akki**
https://issues.apache.org/jira/browse/AIRFLOW-4763 - in case the issue seems valid to the maintainers, here is the fix.

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Review by feluelle — CHANGES_REQUESTED** [review]
What do you think of..
```python
def get_command(self):
        if self.command is not None and isinstance(self.command, str) and self.command.strip().find('[') == 0:
            commands = ast.literal_eval(self.command)
        else:
            commands = self.command
```
Also see [this ](https://stackoverflow.com/questions/26544091/checking-if-type-list-in-python)why I used `isinstance` instead of `type`.

**Comment by akki** [comment]
@feluelle Ahh, good point - just realised the redundancy. But I also see that the `self.command is not None` part is redundant as well in the newly suggested implemented `self.command is not None and isinstance(self.command, str) and self.command.strip().find('[') == 0`, so I am finally changing this to:

```python
def get_command(self):
        if isinstance(self.command, str) and self.command.strip().find('[') == 0:
            commands = ast.literal_eval(self.command)
        else:
            commands = self.command
```

Please have a look and let me know if you see any concerns. Thanks for the review. :)

**Review by feluelle — APPROVED** [review]
LGTM

**Comment by akki** [comment]
@feluelle Thanks for the approval. I just wanted to confirm that this PR is not waiting for any action from my side.
This is my first commit to Airflow so can you please let me know if I have missed some step - sorry for the delay.

**Comment by feluelle** [comment]
Hey @akki you didn't miss any steps. It just needs to be reviewed by a committer. Maybe @kaxil or @mik-laj has some time to review it.

File: airflow/operators/docker_operator.py
Comment (kaxil) [inline airflow/operators/docker_operator.py:250]: Why do we have this `and self.command.strip().find('[') == 0`?

Just trying to understand the logic.

Comment (mik-laj) [reply airflow/operators/docker_operator.py:250]: When Jinja returns the "[" sign, an array is created. Please look at related PR: https://github.com/apache/airflow/pull/4900

Comment (kaxil) [reply airflow/operators/docker_operator.py:250]: Jinja would convert a list to list. `self.command.strip().find('[') == 0` is checking a **string**, right? and checking if the first character is `[`.

Sorry, maybe I am missing something. @mik-laj Feel free to merge this PR :)

Comment (mik-laj) [reply airflow/operators/docker_operator.py:250]: When a string is encountered, but the string looks like it contains arrays (starts with the character "[") it is processed by literal_parse, which creates the real table from the string.
```
>>> import ast
>>> ast.literal_eval('["A", "B"]')
['A', 'B']
>>>
```

Currently, Jinja always accepts a string at the input and returns a string. However, the mentioned PR discussed an alternative solution to the one used in this PR.

I can not do it at the moment. I just wanted to answer your question.

Comment (akki) [reply airflow/operators/docker_operator.py:250]: @kaxil @mik-laj So would you like to see any changes in this PR or does it have your approval?

**Comment by akki** [comment]
Hi, does anyone have any suggestion for this PR?
Else, would any of the committers approve/merge it?

File: airflow/operators/docker_operator.py
Comment (kaxil) [reply airflow/operators/docker_operator.py:250]: Thanks, @mik-laj for the explanation. I understand it now.

**Review by kaxil — APPROVED** [review]
LGTM

**Comment by kaxil** [comment]
Thank you @akki for your contribution and apologies for having to wait this long.

**Comment by chbrown** [comment]
Ran into this bug on my very first custom DAG after going through the tutorials 😣 thanks @akki 👏

FWIW here's my monkeypatch workaround (until the better fix in this PR is released... `1.10.5`?):
```python
from functools import partialmethod
from airflow.operators.docker_operator import DockerOperator
def get_command(operator):
    return operator.command  # assumes command is already a proper list
DockerOperator.get_command = partialmethod(get_command)
```

**Pull request merged**
